src/app/users/users-achievements/users-achievements.service.ts

```diff
@@ -1,19 +1,44 @@
 import { Injectable } from '@angular/core';
 import { CouchService } from '../../shared/couchdb.service';
+import { map } from 'rxjs/operators';
+import { ResourcesService } from '../../resources/resources.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Injectable({
   providedIn: 'root'
 })
 export class UsersAchievementsService {
 
   readonly dbName = 'achievements';
+  resourceIds = [];
+  parent = this.route.snapshot.data.parent;
 
   constructor(
-    private couchService: CouchService
-  ) {}
+    private couchService: CouchService,
+    private resourcesService: ResourcesService,
+    private route: ActivatedRoute
+  ) {
+    this.resourcesService.resourcesListener(this.parent).pipe(map((resources: any) => {
+      resources.map((res: any) => { this.resourceIds.push(res._id); });
+      return this.resourceIds;
```

Comment:
You don't need the `map` function here.  You can simply run all of the code in the `subscribe` function